Tribler/Core/Modules/restapi/channels/channels_subscription_endpoint.py

```diff
@@ -84,10 +84,8 @@ def render_PUT(self, request):
         """
         request.setHeader('Content-Type', 'text/json')
         channel_info = self.get_channel_from_db(self.cid)
-        if channel_info is None:
-            return ChannelsModifySubscriptionEndpoint.return_404(request)
 
-        if channel_info[7] == VOTE_SUBSCRIBE:
+        if channel_info is not None and channel_info[7] == VOTE_SUBSCRIBE:
```

Comment:
No, if the `channel_info` is `None`, the channel is not discovered and this if-statement will never be entered.

I've added the first part of the if-statment to make the 2nd part after the `and` not crash if the channel is not discovered.